public/docs/ts/latest/guide/structural-directives.jade

```diff
@@ -0,0 +1,638 @@
+
+:marked
+  Our directive is quite simple, but surely the real `ng-if` is much more
+  complex? We can find out quite easily by 
+  [looking at the source code](https://github.com/angular/angular/blob/master/modules/angular2/src/common/directives/ng_if.ts).
+  
+  The source code for Angular 2 is quite well documented and we shouldn't be shy
+  about consulting it when we want to know how something works.
+  
+  It turns out that `ng-if` isn't much different at all! There's a few
+  additional checks for performance (avoiding clearing or recreating the
+  view when not needed) but otherwise it's pretty much the same.
+  
+.l-main-section  
+:marked
+  ## The code so far
+  
+  As always, code is king in any guide. Here is the full code used for the
+  examples we've gone through. 
+
++makeTabs(`
+  structural-directives/ts/src/app/unless.directive.ts,
+  structural-directives/ts/src/app/heavy-loader.component.ts,  
+  structural-directives/ts/src/app/structural-directives.component.ts,
+  structural-directives/ts/src/app/structural-directives.component.html
+  `, 
+  null, 
+  `unless.directive.ts,
+   heavy-loader.component.ts, 
+   structural-directives.component.ts,
+   structural-directives.component.html
+   `)
+
+.l-main-section  
+:marked
+  ## Configurable component DOM
+  
+  We've seen how a directive can use a template to stamp out copies of it in 
+  place, but there are more ways in which we can manipulate the DOM.
+  
+  When *declaring* a component in the DOM, Angular 2 will usually disregard
+  whatever its contents since that component will render its own template.
+  
+  For instance, if consider this simple component.
+
++makeExample('structural-directives/ts/src/app/my-component.component.ts')(format=".")
+
+:marked
+  Imagine we were to declare it with some content inside. What happens?
+
++makeExample('structural-directives/ts/src/app/structural-directives.component.html', 'my-component-1')(format=".")
+
+:marked
+  Angular removes the DOM elements between the `my-component` tags and
+  inserts the rendered component template:
+
+figure.image-display
+  img(src='/resources/images/devguide/structural-directives/my-component-content.png' alt="my-component content")
+
+:marked
+  Clearly we want Angular to **incorporate** the material between the tags
+  *within* the component's template. 
+  
+  This is a powerful idea. We could create reusable components that
+  manipulate external DOM elements, oblivious to their representation. We could also provide
+  a component with a template with which to render certain data within itself.  
+  
+  Angular 2 provides the mechanisms to do just this. Over the next sections
+  we'll see how to build such a configurable component that receives from the 
+  outside a great deal of source material that it can manipulate and re-display.
+  
+  We'll focus on three main points:
+  
+  * How to transclude elements in the component declaration
```

Comment:
I'd like to avoid the word "transclude" here since in Angular 2, we're using a mechanism called "content projection", which is based on the Shadow DOM standard. Transclusion is what happend in A1. Whereas projection really just "projects" the contents from light DOM in the (emulated) Shadow DOM, but technically, it's still part of the light DOM.

To be very specific, actually as along as we use Shadow DOM emulation (which is the default), we rather transclude than project, but I'd go with projection tho.